src/java.base/share/classes/com/sun/crypto/provider/PKCS12PBECipherCore.java

```diff
@@ -115,7 +114,7 @@ static byte[] derive(char[] chars, byte[] salt, int ic, int n, int type,
                 hashSupported = false;
             }
             if (hashSupported) {
-                if (nativeCrypto.PBEDerive(passwd, passwd.length, salt, salt.length, key, ic, n, type, hashIndex) != -1) {
+                if (NativeCrypto.getNativeCrypto().PBEDerive(passwd, passwd.length, salt, salt.length, key, ic, n, type, hashIndex) != -1) {
```

Comment:
The call to `NativeCrypto.getNativeCrypto()` is sufficiently expensive that I believe the static field should be kept. Likewise in other classes which access the instance.

Reply:
The static fields are restored.